hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/kafka/SqlPojoTest.java

```diff
@@ -263,11 +261,7 @@ public void test_topLevelFieldExtraction() {
 
     @Test
     public void test_customType() {
```

Comment:
is it legal to create UDT without fields and without options? such type would match anything what underlying mapping contains.

Reply:
It was legal to create UDTs without fields, but they were tied to a serialization format, so could not be created without options. After decoupling the serialization format from UDTs, it has become possible to create a UDT without fields or options. (Recall that type metadata is optional unlike mapping metadata.) As you pointed out, such type can match any format and schema. However, there is a caveat: `FieldEnricher` updates the `Type` with the resolved fields and propagates this change across members by updating the `Type` in `RelationsStorage`. Thus, after the first use of such a UDT, fields are materialized and "match anything" feature is disabled. This is, I think, counterintuitive and we should instead skip materializing fields in `Type`, i.e. "match anything" behavior should be left intact.